Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -72,15 +82,28 @@ sub _releases {
 	my %releaseTypes;
 	my %contributions;
 	my %isPrimaryArtist;
+	my %albumList;
 	foreach (@$albums) {
+		my $genreMatch = undef;
+		if ( $prefs->get('useComposerAsArtist')==2 ) {
+			my $requestRef = [ 'genres', 0, MAX_ALBUMS, "album_id:".$_->{id} ];
+			my $request = Slim::Control::Request->new( $client ? $client->id() : undef, $requestRef );
+			$request->execute();
+			$log->error($request->getStatusText()) if $request->isStatusError();
+			my $genres= $request->getResult('genres_loop');
+			foreach my $genre (@$genres) {
+				$genreMatch = grep {uc($genre->{genre}) eq $_} split(/,/, uc($prefs->get('useComposerAsArtistGenres'))) if !$genreMatch;
+			}
+		}
```

Comment:
This might not be a problem in most cases. But running an additional query for each album is potentially expensive. Did you look into adding genre info to the result of the `albums` query? That would be much more elegant. IMHO anyway.

Reply:
I did have a look at that, but it would involve a subquery to retrieve potentially multiple genres for the album, so six of one and half a dozen of the other, I think.

Reply:
Fair enough.